src/nvim/eval/decode.c

```diff
@@ -973,7 +973,13 @@ int msgpack_to_vim(const msgpack_object mobj, typval_T *const rettv)
       }
       break;
     }
-    case MSGPACK_OBJECT_FLOAT: {
+#ifdef NVIM_MSGPACK_HAS_FLOAT32
```

Comment:
Can't we test `MSGPACK_VERSION_*` instead? https://github.com/msgpack/msgpack-c/blob/master/include/msgpack/version.h

That also avoids the extra cmake code.